torch/distributed/fsdp/_init_utils.py

```diff
@@ -359,8 +361,12 @@ def _init_param_handle_from_params(
     state._handles.append(handle)
     for module in handle.flat_param._modules:
         state._module_to_handles[module].append(handle)
-    for module in handle.flat_param._root_modules:
-        state._root_module_to_handles[module].append(handle)
+    state._comm_module_to_handles[handle._comm_module].append(handle)
+    num_comm_module_handles = len(state._comm_module_to_handles[handle._comm_module])
+    assert num_comm_module_handles == 1, (
```

Comment:
Curious, since append was only called once on `_comm_module_to_handles` and the loop calling `_init_param_handle_from_params ` is keyed on comm module, when do we expect this assert be violated?

Reply:
This assert cannot be violated with today's code, but if we did change the `FlatParameter` construction algorithm (say for grouping sibling modules together in the future), then we have this assert here to remind us that the current implementation assumes this.